### PR TITLE
Wrap ByronBlock in the HFC

### PR DESCRIPTION
**Pull request by mrBliss**

### CODE REVIEW

**Comment by dcoutts** [comment]
Could I include this one-liner into a patch in this same PR?
```
 module Ouroboros.Consensus.HardFork.Combinator.Ledger (
     HardForkLedgerError(..)
   , HardForkEnvelopeErr(..)
+  , HardForkLedgerWarning(..)
```
Needed for the node update to be able to trace these warnings.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+instance NoHardForks ByronBlock where
+  getEraParams cfg =
+      byronEraParams
+        HardFork.NoLowerBound
+        (byronGenesisConfig (configBlock cfg))
+  toPartialConsensusConfig _ = id
+  toPartialLedgerConfig _ cfg = ByronPartialLedgerConfig {
+        byronLedgerConfig = cfg
+      , triggerHardFork   = undefined
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs:108]: Use `TriggerHardForkNever`

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+pattern ByronHFCNodeToNodeVersion2 :: HardForkNodeToNodeVersion '[ByronBlock]
+pattern ByronHFCNodeToNodeVersion2 = HardForkNodeToNodeDisabled ByronNodeToNodeVersion2
+
+pattern ByronHFCNodeToClientVersion1 :: HardForkNodeToClientVersion '[ByronBlock]
+pattern ByronHFCNodeToClientVersion1 = HardForkNodeToClientDisabled ByronNodeToClientVersion1
+
+instance SupportedNetworkProtocolVersion ByronBlockHFC where
+  supportedNodeToNodeVersions _ = Map.fromList [
+        (NodeToNodeV_1, ByronHFCNodeToNodeVersion1)
+        -- TODO enable V_2?
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs:127]: We can't

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      ]
+
+  supportedNodeToClientVersions _ = Map.fromList [
+        (NodeToClientV_1, ByronHFCNodeToClientVersion1)
+      , (NodeToClientV_2, ByronHFCNodeToClientVersion1)
+        -- TODO enable V_3?
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs:133]: We can't

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        reconstructNestedCtxt (Proxy @(Header ByronBlock)) prefix blockSize
+
+{-------------------------------------------------------------------------------
+  Project TopLevelConfig
+-------------------------------------------------------------------------------}
+
+-- | We are lucky that for Byron we can construct all the full configs from
+-- the partial ones, which means we can reconstruct the 'TopLevelConfig' for
+-- Byron.
+projTopLevelConfig
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/ByronHFC.hs:159]: You can just use `project` :facepalm:

**Comment by mrBliss** [comment]
> Could I include this one-liner into a patch in this same PR?
> 
> ```
>  module Ouroboros.Consensus.HardFork.Combinator.Ledger (
>      HardForkLedgerError(..)
>    , HardForkEnvelopeErr(..)
> +  , HardForkLedgerWarning(..)
> ```
> 
> Needed for the node update to be able to trace these warnings.

See https://github.com/input-output-hk/ouroboros-network/pull/2405

**Comment by mrBliss** [comment]
GitHub is not seeing force-pushes atm, so closing in favour of #2414.

**Pull request closed without merging**